Add Windows ARM64 support

dennisameling opened the pull request.

dennisameling committed:
Add Windows ARM64 support

dscho commented on example/console-example/WinToast Console Example.vcxproj:
```diff
@@ -17,18 +17,26 @@
     <ProjectGuid>{7C5AC60D-8668-47B6-9D05-FB363F854065}</ProjectGuid>
     <Keyword>Win32Proj</Keyword>
-    <WindowsTargetPlatformVersion>8.1</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
```
Will this change anything with regard to Windows 8 support?

If so, would it make sense to override that platform in a `<PropertyGroup Include="*|ARM64">`-like section instead?

dennisameling replied:
It shouldn't break Windows 8 support. The [upstream repo](https://github.com/mohabouje/WinToast/commit/017cf350c1350a3ff2afabe95bfdf89d9e73da05) also uses this TargetPlatform already. I'll build on a Windows 11 machine and will then test the resulting binary on a Windows 8.1 VM (currently downloading an ISO for it).

Alovchin91 replied:
`WindowsTargetPlatformVersion`, despite its name, specifies the version of the SDK to use. 10.x SDK does support Windows 8.1: https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/

dennisameling replied:
@dscho I just tested this on Windows 8.1. The version I created using the Windows 10.x SDK and PlatformToolset=v143 is working "correctly". That is, it has the exact same behavior as the version that comes with GfW today.

Current version (unmodified by me) that pops up when installing GfW 2.38.0 and running the `Git for Windows Updater` scheduled task:

<img width="293" alt="image" src="https://user-images.githubusercontent.com/17739158/199188366-c6197ceb-e660-4fdc-8abc-9fff10b940c6.png">

The version I created using the toolset mentioned above (binaries can be found [here](https://github.com/dennisameling/WinToast/releases/tag/test-dennis)):

```
$ "./WinToast.Console.Example.x64.exe" --appname "Git for Windows" \
--appid GitForWindows.Updater \
--image /mingw64/share/git/git-for-windows.ico \
--text "Download and install Git for Windows DEMO?" \
--action Yes --action No --expirems 15000
```

<img width="286" alt="image" src="https://user-images.githubusercontent.com/17739158/199189211-a0db18af-c507-42ba-98f4-a092cd983d79.png">

So for the scope of this PR, everything works correctly. However, it looks like WinToast can't deal with .ico files on Windows 8.1. When I invoke the toast with a quick and dirty `.png` file, the logo is shown correctly: 

<img width="302" alt="image" src="https://user-images.githubusercontent.com/17739158/199189539-aea688d0-1fd6-4921-8427-431ed3c67ef4.png">

Is that something you want to have fixed at some point? I could provide a follow-up PR in [build-extra](https://github.com/git-for-windows/build-extra/blob/afbd82e9475f1f9f81760f6100e2d71d829c04c5/git-extra/git-update-git-for-windows#L297-L301) to fix the logo thing, but would need a proper `.png` version of the logo first.

Thanks!

rimrul replied:
Seeing as the unmodified binary produces the same result, I'd say the Icon is probably unrelated to this PR.

dennisameling replied:
Unrelated indeed. Just wanted to mention it here to make you aware 👍🏼

dscho replied:
@dennisameling thank you for digging deeper. I agree that it is unrelated, and if you find the time to open that follow-up PR in build-extra, I will be delighted to merge it!

dscho commented on example/console-example/WinToast Console Example.vcxproj:
```diff
@@ -17,18 +17,26 @@
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v140</PlatformToolset>
+    <PlatformToolset>v143</PlatformToolset>
```
Same comment here: would this break Windows 8 support?

Alovchin91 replied:
This only specifies the version of the toolset, e.g. the compiler. The toolset still supports Windows 8.1: https://learn.microsoft.com/en-us/cpp/overview/supported-platforms-visual-cpp?view=msvc-170

dscho committed:
Merge branch 'main' into windows-arm64

dscho committed:
ci: add a CI build

This is loosely based on the one from
https://github.com/git-for-windows/7-Zip

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
